solidity/test/rewards/TestECDSARewards.js

```diff
@@ -0,0 +1,232 @@
+const {accounts, contract, web3} = require("@openzeppelin/test-environment")
+const {createSnapshot, restoreSnapshot} = require("../helpers/snapshot")
+const {initialize, fund, createMembers} = require("./rewardsSetup")
+
+const {expectRevert, time} = require("@openzeppelin/test-helpers")
+
+const BondedECDSAKeepStub = contract.fromArtifact("BondedECDSAKeepStub")
+const ECDSARewards = contract.fromArtifact("ECDSARewards")
+
+const BN = web3.utils.BN
+
+const chai = require("chai")
+chai.use(require("bn-chai")(BN))
+const expect = chai.expect
+
+describe("ECDSARewards", () => {
+  let rewardsContract
+  let keepToken
+
+  let tokenStaking
+  let keepFactory
+  let keepMembers
+  let expectedKEEPAllocation
+
+  const owner = accounts[0]
+
+  // Solidity is not very good when it comes to floating point precision,
+  // we are allowing for ~1 KEEP difference margin between expected and
+  // actual value.
+  const precision = 1
+  const tokenDecimalMultiplier = web3.utils.toBN(10).pow(web3.utils.toBN(18))
+  const firstKeepCreationTimestamp = 1600041600 // Sep 14 2020
+
+  // 1,000,000,000 - total KEEP supply
+  //   200,000,000 - 20% of the total supply goes to staker rewards
+  //   180,000,000 - 90% of staker rewards goes to the ECDSA staker rewards
+  //   178,200,000 - 99% of ECDSA staker rewards goes to keeps opened after Sep 14 2020
+  const KEEPRewards = web3.utils.toBN(178200000).mul(tokenDecimalMultiplier)
+
+  before(async () => {
+    const setup = await initialize()
+    tokenStaking = setup.tokenStaking
+    keepToken = setup.keepToken
+    keepFactory = setup.keepFactory
+
+    keepMembers = await createMembers(tokenStaking)
+    rewardsContract = await ECDSARewards.new(
+      keepToken.address,
+      keepFactory.address,
+      {from: owner}
+    )
+
+    await fund(keepToken, rewardsContract, KEEPRewards)
+  })
+
+  beforeEach(async () => {
+    await createSnapshot()
+  })
+
+  afterEach(async () => {
+    await restoreSnapshot()
+  })
+
+  describe("interval allocation", async () => {
+    expectedKEEPAllocation = [
+      7128000.0, 13685760.0, 15738624.0, 16997713.92, 18697485.31, 15892862.52,
+      13508933.14, 11482593.17, 9760204.19, 8296173.56, 7051747.53, 5993985.4,
+      5094887.59, 4330654.45, 3681056.28, 3128897.84, 2659563.16, 2260628.69,
+      1921534.39, 1633304.23, 1388308.59, 1180062.31, 1003052.96, 852595.02
+    ]
+
+    it("should equal to expected allocations when 5 keeps were created per interval", async () => {
+      await verifyIntervalAllocations(5)
+    })
+
+    it("should equal to expected allocations when 1 keep was created per interval", async () => {
+      await verifyIntervalAllocations(1)
```

Comment:
Shouldn't we have a failing test here? `minimumECDSAKeepsPerInterval = 4` right now and we have only one keep per interval. Allocations compared to the previous test should be 4 times smaller.